src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,156 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+            self.__installation_patches_copy = self.__installation_substatus_msg_copy['patches']
+            low_pri_index = self.__get_installation_low_pri_index(self.__installation_patches_copy)
+
+        status_file_without_patches_size_in_bytes = self.__size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # Deepcopy, fully copies the object to avoid reference modifications
+
+        max_allowed_patches_size_in_bytes = Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES - status_file_without_patches_size_in_bytes
+        self.composite_logger.log_debug("Status file limits evaluated. [FileSizeWithoutPatchesInBytes={0}] [MaxAllowedPatchesSizeInBytes={1}]".format(
+            str(status_file_without_patches_size_in_bytes), str(max_allowed_patches_size_in_bytes)))
+
+        while status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:
+            # Start truncation process
+            patches_retained_in_assessment, self.__assessment_patches_removed, patches_retained_in_installation, self.__installation_patches_removed = \
+                self.__start_truncation_process(self.__assessment_patches_copy, self.__installation_patches_copy, max_allowed_patches_size_in_bytes, low_pri_index)
+
+            if len(self.__assessment_patches_removed) > 0:
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_ASSESSMENT_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_patches=patches_retained_in_assessment,
+                    substatus_message=self.__assessment_substatus_msg_copy, substatus_index=assessment_substatus_index)
+
+            if len(self.__installation_patches_removed) > 0:
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_INSTALLATION_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_patches=patches_retained_in_installation,
+                    substatus_message=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_size_diff_in_bytes = (status_file_size_in_bytes - Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)  # The byte difference is tombstone, new errors, and escape chars byte size
+            max_allowed_patches_size_in_bytes -= status_file_size_diff_in_bytes   # Reduce the max patches byte size by tombstone, new errors, and escape chars byte size
+
```

Comment:
change comment to: Computing max patch size allowed post truncation.

Going back to the discussion on maintaining exclusivity in PRs. The part on tombstone and error details will come in their individual PRs

Reply:
After going through this again, as stated in the other comment, please remove lines 905-906

Reply:
removed, done